test/src/test/java/jenkins/model/NodesTest.java

```diff
@@ -88,6 +96,92 @@ public void addNodeShouldReplaceExistingNode() throws Exception {
         assertThat(r.jenkins.getNode("foo"), sameInstance(newNode));
     }
 
+    @Test
+    public void removeNodeWithNonIdleExecutorsTest() throws Exception {
+        Node node = r.createOnlineSlave();
+        r.jenkins.addNode(node);
+        TestFlyweightTask task = new TestFlyweightTask();
+        task.label = node.getSelfLabel();
+        r.jenkins.getQueue().schedule(task,0);
+        //wait for execution
+        while(!task.isBuilding){
```

Comment:
Would be probably slightly better to use `awaitility` for such async "waits/retries until".
See https://github.com/jenkinsci/jenkins/blob/master/test/src/test/java/hudson/cli/QuietDownCommandTest.java#L460-L461 for an example.

Here this would something like:

```java
await().atMost(5, TimeUnit.SECONDS)
       .until(task::isBuilding);
```